### PR TITLE
Migrate add and edit incoming webhook components to redux

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
Migrate add and edit incoming webhook components to redux.
Also fixed some status codes for creating outgoing webhooks.

#### Ticket Link
GitHub for Add: https://github.com/mattermost/platform/issues/6442
GitHub for Edit: https://github.com/mattermost/platform/issues/6446

#### Checklist
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: webapp/components/integrations/components/add_incoming_webhook/add_incoming_webhook.jsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+export default class AddIncomingWebhook extends React.PureComponent {
+    static propTypes = {
+
+        /**
+        * The current team
+        */
+        team: PropTypes.object.isRequired,
+
+        /**
+        * The request state for createOutgoingHook action. Contains status and error
```
Comment (jwilander) [inline webapp/components/integrations/components/add_incoming_webhook/add_incoming_webhook.jsx]: Copy/paste error should be `createIncomingHook`

File: webapp/components/integrations/components/add_incoming_webhook/add_incoming_webhook.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        */
+        createIncomingHookRequest: PropTypes.object.isRequired,
+
+        actions: PropTypes.shape({
+
+            /**
+            * The function to call to add a new outgoing webhook
```
Comment (jwilander) [inline webapp/components/integrations/components/add_incoming_webhook/add_incoming_webhook.jsx]: Here as well

File: webapp/components/integrations/components/edit_incoming_webhook/edit_incoming_webhook.jsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        */
+        hook: PropTypes.object,
+
+        /**
+        * The id of the outgoing webhook to edit
+        */
+        hookId: PropTypes.string.isRequired,
+
+        /**
+        * The request state for updateOutgoingHook action. Contains status and error
```
Comment (jwilander) [inline webapp/components/integrations/components/edit_incoming_webhook/edit_incoming_webhook.jsx]: Here and a couple below as well :)

File: webapp/components/integrations/components/add_incoming_webhook/add_incoming_webhook.jsx (hunk 1 above)
Comment (cpanato) [inline webapp/components/integrations/components/add_incoming_webhook/add_incoming_webhook.jsx]: my bad :(
done

File: webapp/components/integrations/components/add_incoming_webhook/add_incoming_webhook.jsx (hunk 2 above)
Comment (cpanato) [inline webapp/components/integrations/components/add_incoming_webhook/add_incoming_webhook.jsx]: done

File: webapp/components/integrations/components/edit_incoming_webhook/edit_incoming_webhook.jsx (hunk 3 above)
Comment (cpanato) [inline webapp/components/integrations/components/edit_incoming_webhook/edit_incoming_webhook.jsx]: done

**Pull request merged**